test/Vegetation/multi_layer_root_test.jl

```diff
@@ -0,0 +1,119 @@
+    LAI = FT(1)
+    area_index = (root = RAI, stem = SAI, leaf = LAI)
+    h_stem = FT(3.0)
+    h_leaf = FT(3.0)
+    function K_sat_root(z::T) where {T}
+        return FT(1.8e-8 * abs(z)) # on slow end of plant conductance range
+    end
+    K_sat_stem = FT(1.8e-8) 
+    K_sat_leaf = FT(1.8e-8) 
+    above_ground_plant_K_sat = (stem = K_sat_stem, leaf = K_sat_leaf) 
+    plant_vg_α = FT(0.002)
+    plant_vg_n = FT(4.2)
+    plant_vg_m = FT(1) - FT(1) / plant_vg_n
+    plant_ν = FT(0.7)
+    plant_S_s = FT(1e-2 * 0.0098) # m3/m3/MPa to m3/m3/m
+    root_depths = -Array(10:-1:1.0) ./ 10.0 * 2.0 .+ 0.2 / 2.0
+    n_stem = Int64(1)
+    n_leaf = Int64(1)
+    compartment_midpoints = [h_stem/2, h_stem+h_leaf/2]
+    compartment_surfaces = [FT(0.0), h_stem, h_stem+h_leaf]
+    function root_distribution(z::T) where {T}
+        return T(1.0 / 0.5) * exp(z / T(0.5)) 
+    end  
+    earth_param_set = create_lsm_parameters(FT)
+    plant_hydraulics_domain =
+            PlantHydraulicsDomain(root_depths,n_stem,n_leaf,compartment_midpoints,compartment_surfaces)
+    plant_hydraulics_ps =
+        PlantHydraulics.PlantHydraulicsParameters{FT, typeof(earth_param_set)}(
+            area_index,
+            above_ground_plant_K_sat,
+            K_sat_root,
+            plant_vg_α,
+            plant_vg_n,
+            plant_vg_m,
+            plant_ν,
+            plant_S_s,
+            root_distribution,
+            earth_param_set,
+        )      
+
+    function leaf_transpiration(t::ft) where {ft}
+        T = ft(0.0)
+        return T
+    end
+
+    p_soil0 = FT.((-2.0 .- root_depths) .* 9800)
+    transpiration = PrescribedTranspiration{FT}((t::FT) -> leaf_transpiration(t))
+    root_extraction = PrescribedSoilPressure{FT}((t::FT) -> p_soil0)
+
+    plant_hydraulics = PlantHydraulics.PlantHydraulicsModel{FT}(;
+        domain = plant_hydraulics_domain,
+        parameters = plant_hydraulics_ps,
+        root_extraction = root_extraction,
+        transpiration = transpiration
+    )
+
+    Y, p, cds = initialize(plant_hydraulics)
+    ode! = make_ode_function(plant_hydraulics)
+
+    p_stem_ini = FT(-0.1e6 / 9800) # Pa to m 
+    p_leaf_ini = FT(-0.2e6 / 9800)
+
+    S_l_ini = inverse_water_retention_curve.(
+        plant_vg_α,
+        plant_vg_n,
+        plant_vg_m,
+        [p_stem_ini, p_leaf_ini],
+        plant_ν,
+        plant_S_s) 
+
+    Y.vegetation.ϑ_l .= augmented_liquid_fraction.(plant_ν, S_l_ini)
+
+    update_aux! = make_update_aux(plant_hydraulics)
+    update_aux!(p,Y,0.0)
+
+    t0 = FT(0)
+    N_days = 30
+    tf = FT(3600 * 24 * N_days)
+    dt = FT(100)
+
+    sv = SavedValues(FT, ClimaCore.Fields.FieldVector)
+    daily = Array(2:(3600 * 24):(N_days * 3600 * 24))
+    prob = ODEProblem(ode!, Y, (t0, tf), p);
+    cb = SavingCallback(
+        (u, t, integrator) -> copy(integrator.p),
+        sv;
+        saveat = daily,
+    )
+    sol = solve(prob, RK4(); dt = dt, saveat = daily, callback = cb, adaptive=false) 
```

Comment:
Im not sure this is a very good test. Maybe instead of solving an integration, we can test the rhs function directly? What do you want to show with this test?

Reply:
can the plant_hydraulics_test incorporate a larger number of layers and supersede this?

Reply:
I will include the larger number of layers in plant_hydraulics_test - was just wanting to test the multiple layers

Reply:
did this, now deleting multi layer root test